### ISSUE TITLE
BUG : Get video fps=0

### ISSUE THREAD

**Issue by helyski** [helyski/android-libavtranscoder#3]
Some TS file will get FPS=0.
```
hw_decode :: analysis stream[1] ....................
hw_decode :: ... ... AVStream ... ...
hw_decode :: AVStream->start_time:8650
hw_decode :: AVStream->duration:5405097
hw_decode :: AVStream->time_base.num:1
hw_decode :: AVStream->time_base.den:90000
hw_decode :: AVStream->avg_frame_rate.num:0
hw_decode :: AVStream->avg_frame_rate.den:0
hw_decode :: AVStream->r_frame_rate.num:25
hw_decode :: AVStream->r_frame_rate.den:1
hw_decode ::  ... ... AVCodexContext ... ...
hw_decode :: AVCodecContext->time_base.num:1
hw_decode :: AVCodecContext->time_base.den:180000
hw_decode :: AVCodecContext->framerate.den:0
hw_decode :: AVCodecContext->framerate.num:0
hw_decode :: AVCodecContext->codec_id:27
hw_decode :: AVCodecContext->codec_type:0
hw_decode :: AVCodecContext->width:1280
hw_decode :: AVCodecContext->height:720
hw_decode :: AVCodecContext->gop_size:12
hw_decode :: AVCodecContext->bitrate:0
hw_decode :: AVCodecContext->extradata_size:32
```

**Comment by helyski** [comment]
# Try to get FPS from:

1. AVStream->avg_frame_rate.
2. AVStream->r_frame_rate.
3. AVStream->(AVCodecContext*)codec->framerate.

# FIX IT

update the code in decode_tool->get_decode_info_from_input:
```
 // Get FPS from AVStream avg_frame_rate.
 if(st->avg_frame_rate.den > 0) {
     pVideoInfo->fps = st->avg_frame_rate.num / st->avg_frame_rate.den;
 }
 // Get FPS from AVStream r_frame_rate.
 if(pVideoInfo->fps <=0) {
     if(st->r_frame_rate.den > 0) {
         pVideoInfo->fps = st->r_frame_rate.num / st->r_frame_rate.den;
     }
 }
 if(pVideoInfo->fps <=0) {
     if (st->codec->framerate.den > 0) {
         pVideoInfo->fps = st->codec->framerate.num / st->codec->framerate.den;
     }
 }
```

**Comment by helyski** [comment]
Resolved in commit:db54ec413216c87e0aa33c38c8e9c9a91100c282
